### PR TITLE
[19.01] Fix hidden parameter use in workflows

### LINKED ISSUE #7454: WF is terminated on 19.01
Hello,

Having GraphClust2 docker tested with Galaxy 19.01, I encounter the premature termination of the workflows. 
The error message seems to be : `MessageException: Error due to input mapping of 'NSPDK_candidateClusters' in 'iteration_num|CI'.  A common cause of this is conditional outputs that cannot be determined until runtime, please review your workflow.`

Related tools repo: https://github.com/bgruening/galaxytools/tree/master/tools/GraphClust/NSPDK

Some extended log:
```
dk_sparse/9.2.3' is: handlers                                                                                                                                                                                                                 
galaxy.util.handlers DEBUG 2019-03-04 10:57:40,393 (Job[unflushed,tool_id=toolshed.g2.bx.psu.edu/repos/rnateam/graphclust_nspdk/nspdk_sparse/9.2.3]) Selected handler 'handler0' by random choice from handler tag 'handlers'                 
                                                                                                                                                                                                                                              
==> /home/galaxy/logs/uwsgi.log <==                                                                                                                                                                                                           
132.230.153.74 - - [04/Mar/2019:10:57:39 +0000] "GET /api/histories/f597429621d6eb2b/contents?order=hid&v=dev&q=update_time-ge&q=deleted&q=purged&qv=2019-03-04T10%3A57%3A35.000Z&qv=False&qv=False HTTP/1.1" 200 - "http://192.52.32.222:8080
/workflows/run?id=f2db41e1fa331b3e" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.96 Safari/537.36"                                                                                                
[pid: 324|app: 0|req: 243/486] 132.230.153.74 () {44 vars in 1120 bytes} [Mon Mar  4 10:57:39 2019] GET /api/histories/f597429621d6eb2b/contents?order=hid&v=dev&q=update_time-ge&q=deleted&q=purged&qv=2019-03-04T10%3A57%3A35.000Z&qv=False&
qv=False => generated 2 bytes in 57 msecs (HTTP/1.1 200) 3 headers in 124 bytes (1 switches on core 1)                                                                                                                                        
132.230.153.74 - - [04/Mar/2019:10:57:40 +0000] "GET /api/histories/f597429621d6eb2b?keys=size%2Cnon_ready_jobs%2Ccontents_active%2Chid_counter HTTP/1.1" 200 - "http://192.52.32.222:8080/workflows/run?id=f2db41e1fa331b3e" "Mozilla/5.0 (X1
1; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.96 Safari/537.36"                                                                                                                                                    
[pid: 324|app: 0|req: 244/487] 132.230.153.74 () {44 vars in 1016 bytes} [Mon Mar  4 10:57:40 2019] GET /api/histories/f597429621d6eb2b?keys=size%2Cnon_ready_jobs%2Ccontents_active%2Chid_counter => generated 116 bytes in 102 msecs (HTTP/1
.1 200) 3 headers in 124 bytes (1 switches on core 0)                                                                                                                                                                                         
                                                                                                                                                                                                                                              
==> /home/galaxy/logs/handler0.log <==                                                                                                                                                                                                        
galaxy.util.handlers INFO 2019-03-04 10:57:40,451 Flushed transaction for Job[id=13,tool_id=toolshed.g2.bx.psu.edu/repos/rnateam/graphclust_nspdk/nspdk_sparse/9.2.3] (58.021 ms)                                                             
galaxy.util.handlers INFO 2019-03-04 10:57:40,451 (Job[id=13,tool_id=toolshed.g2.bx.psu.edu/repos/rnateam/graphclust_nspdk/nspdk_sparse/9.2.3]) Handler 'handler0' assigned using 'db-preassign' assignment method                            
galaxy.tools.execute DEBUG 2019-03-04 10:57:40,451 Tool [toolshed.g2.bx.psu.edu/repos/rnateam/graphclust_nspdk/nspdk_sparse/9.2.3] created job [13] (167.194 ms)                                                                              
galaxy.tools.execute DEBUG 2019-03-04 10:57:40,502 Executed 1 job(s) for tool toolshed.g2.bx.psu.edu/repos/rnateam/graphclust_nspdk/nspdk_sparse/9.2.3 request: (323.947 ms)                                                                  
galaxy.workflow.run DEBUG 2019-03-04 10:57:40,526 Workflow step 5 of invocation 3 invoked (354.480 ms)                                                                                                                                        
galaxy.workflow.run ERROR 2019-03-04 10:57:40,564 Failed to schedule Workflow[id=1,name=GraphClust-MotifFinder], problem occurred on WorkflowStep[index=5,type=tool].                                                                         
Traceback (most recent call last):                                                                                                                                                                                                            
  File "/galaxy-central/lib/galaxy/workflow/run.py", line 190, in invoke                                                                                                                                                                      
    incomplete_or_none = self._invoke_step(workflow_invocation_step)                                                                                                                                                                          
  File "/galaxy-central/lib/galaxy/workflow/run.py", line 266, in _invoke_step                                                                                                                                                                
    use_cached_job=self.workflow_invocation.use_cached_job)                                                                                                                                                                                   
  File "/galaxy-central/lib/galaxy/workflow/modules.py", line 1191, in execute                                                                                                                                                                
    raise exceptions.MessageException(message)                                                                                                                                                                                                
MessageException: Error due to input mapping of 'NSPDK_candidateClusters' in 'iteration_num|CI'.  A common cause of this is conditional outputs that cannot be determined until runtime, please review your workflow.                         
galaxy.workflow.run ERROR 2019-03-04 10:57:40,564 Failed to execute scheduled workflow.                                                                                                                                                       
Traceback (most recent call last):                                                                                                                                                                                                            
  File "/galaxy-central/lib/galaxy/workflow/run.py", line 83, in __invoke                                                                                                                                                                     
    outputs = invoker.invoke()                                                                                                                                                                                                                
  File "/galaxy-central/lib/galaxy/workflow/run.py", line 190, in invoke                                                                                                                                                                      
    incomplete_or_none = self._invoke_step(workflow_invocation_step)                                                                                                                                                                          
  File "/galaxy-central/lib/galaxy/workflow/run.py", line 266, in _invoke_step                                                                                                                                                                
    use_cached_job=self.workflow_invocation.use_cached_job)                                                                                                                                                                                   
  File "/galaxy-central/lib/galaxy/workflow/modules.py", line 1191, in execute                                                                                                                                                                
    raise exceptions.MessageException(message)                                                                                                                                                                                                
MessageException: Error due to input mapping of 'NSPDK_candidateClusters' in 'iteration_num|CI'.  A common cause of this is conditional outputs that cannot be determined until runtime, please review your workflow.                         
galaxy.workflow.scheduling_manager DEBUG 2019-03-04 10:57:40,583 Workflow invocation [3] scheduled                                                                                                                                            
galaxy.jobs.mapper DEBUG 2019-03-04 10:57:41,143 (13) Mapped job to destination id: slurm_cluster                                                                                                                                             
galaxy.jobs.handler DEBUG 2019-03-04 10:57:41,159 (13) Dispatching to slurm runner                                                                                                                                                            
galaxy.jobs DEBUG 2019-03-04 10:57:41,174 (13) Persisting job destination (destination id: slurm_cluster)                                                                                                                                     
galaxy.jobs DEBUG 2019-03-04 10:57:41,176 (13) Working directory for job is: /export/galaxy-central/database/job_working_directory/000/13                                                                                                     
galaxy.jobs.runners DEBUG 2019-03-04 10:57:41,201 Job [13] queued (41.246 ms)                                                                                                                                                                 
galaxy.jobs.handler INFO 2019-03-04 10:57:41,231 (13) Job dispatched                                                                                                                                                                          
galaxy.tools.deps DEBUG 2019-03-04 10:57:41,390 Using dependency graphclust-wrappers version 0.6.0 of type conda                                                                                                                              
galaxy.tools.deps DEBUG 2019-03-04 10:57:41,390 Using dependency nspdk version 9.2 of type conda                                                                                                                                              
galaxy.jobs.command_factory INFO 2019-03-04 10:57:41,414 Built script [/export/galaxy-central/database/job_working_directory/000/13/tool_script.sh] for tool command [[ "$CONDA_DEFAULT_ENV" = "/export/tool_deps/_conda/envs/mulled-v1-f44845
823d7559f16922b92aa5669e869a0464790f169a4df9ef5543cb4474a6" ] ||                                                                                                                                                                              
MAX_TRIES=3                                                                                                                                                                                                                                   
COUNT=0                                                                                                                                                                                                                                       
while [ $COUNT -lt $MAX_TRIES ]; do                                                                                                                                                                                                           
    . /export/tool_deps/_conda/bin/activate '/export/tool_deps/_conda/envs/mulled-v1-f44845823d7559f16922b92aa5669e869a0464790f169a4df9ef5543cb4474a6' > conda_activate.log 2>&1                                                              
    if [ $? -eq 0 ];then                                                                                                                                                                                                                      
        break                                                                                                                                                                                                                                 
    else                                                                                                                                                                                                                                      
        let COUNT=COUNT+1                                                                                                                                                                                                                     
        if [ $COUNT -eq $MAX_TRIES ];then                                                                                                                                                                                                     
            echo "Failed to activate conda environment! Error was:"                                                                                                                                                                           
            cat conda_activate.log                                                                                                                                                                                                            
            exit 1                                                                                                                                                                                                                            
        fi                                                                                                                                                                                                                                    
        sleep 10s                                                                                                                                                                                                                             
    fi                                                                                                                                                                                                                                        
done ; [ "$CONDA_DEFAULT_ENV" = "/export/tool_deps/_conda/envs/mulled-v1-f44845823d7559f16922b92aa5669e869a0464790f169a4df9ef5543cb4474a6" ] ||    
```

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes #7454, broken in https://github.com/galaxyproject/galaxy/commit/dd8d74f364770a1272555a3753f5d1e1288315e1

### CODE REVIEW

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by nsoranzo** [comment]
```
[docker-api] $ /bin/bash /var/jenkins/tmp/jenkins316398275589431027.sh
Running build number 14797
Docker version:
Docker version 17.05.0-ce, build 89658be
Launching docker container for testing with extra args -e GALAXY_TEST_UID=999 -e GALAXY_SKIP_CLIENT_BUILD=1 ...
Unable to find image 'mvdbeek/testing-base:19.01.3' locally
docker: Error response from daemon: manifest for mvdbeek/testing-base:19.01.3 not found.
See 'docker run --help'.
Build step 'Execute shell' marked build as failure
```

**Comment by mvdbeek** [comment]
ouch, that image shouldn't have made it ...

File: run_tests.sh [hunk 1]
```diff
@@ -270,7 +270,7 @@ exists() {
     type "$1" >/dev/null 2>/dev/null
 }
 
-DOCKER_DEFAULT_IMAGE='mvdbeek/testing-base:19.01.3'
```
Comment (martenson) [inline run_tests.sh:273]: so this is how we got this image all over the jenkins infra (thanks for fixing it!)

**Review by jmchilton — APPROVED** [review]

**Pull request merged**